In [1]:
import pandas as pd

In [2]:
netflix_originals = pd.read_csv('netflix_originals.csv')
netflix_titles = pd.read_csv('netflix_titles.csv')

## Análisis Netflix Original. 

In [3]:

netflix_originals = netflix_originals.loc[:, ~netflix_originals.columns.str.contains('^Unnamed')]


In [4]:
pd.set_option("display.max_columns", None)

In [5]:
netflix_originals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 24.2+ KB


In [6]:
# Reemplazar posibles errores en las fechas (por ejemplo, punto por coma)
netflix_originals['Premiere'] = netflix_originals['Premiere'].str.replace(r"\.", ",", regex=True)

# Intentar convertir la columna 'Premiere' a datetime
try:
    netflix_originals['Premiere'] = pd.to_datetime(netflix_originals['Premiere'], format='mixed')
    print("Columna 'Premiere' convertida exitosamente a datetime.")
except Exception as e:
    print(f"Error al convertir 'Premiere': {e}")

Columna 'Premiere' convertida exitosamente a datetime.


In [7]:
print(netflix_originals['Premiere'].head())

0   2019-08-05
1   2020-08-21
2   2019-12-26
3   2018-01-19
4   2020-10-30
Name: Premiere, dtype: datetime64[ns]


In [21]:
netflix_originals.describe()

Runtime  IMDB Score
count  513.000000  513.000000
mean    94.674464    6.210916
std     26.648121    0.968850
min      4.000000    2.500000
25%     87.000000    5.600000
50%     97.000000    6.300000
75%    108.000000    6.900000
max    209.000000    9.000000

In [8]:
netflix_originals.isnull().sum()

Title         0
Genre         0
Premiere      0
Runtime       0
IMDB Score    0
Language      0
dtype: int64

In [24]:
netflix_originals.nunique()

Title         513
Genre         106
Premiere      350
Runtime       113
IMDB Score     52
Language       37
dtype: int64

In [29]:
netflix_originals.dtypes

Title          object
Genre          object
Premiere       object
Runtime         int64
IMDB Score    float64
Language       object
dtype: object

In [ ]:
print(netflix_originals['Title'].head(10))


0                    Enter the Anime
1                        Dark Forces
2                            The App
3                     The Open House
4                        Kaali Khuhi
5                              Drive
6                  Leyla Everlasting
7    The Last Days of American Crime
8                            Paradox
9                 Sardar Ka Grandson
Name: Title, dtype: object


In [32]:
non = netflix_originals.select_dtypes(include='number')
non

Runtime  IMDB Score
0         58         2.5
1         81         2.6
2         79         2.6
3         94         3.2
4         90         3.4
..       ...         ...
508      114         8.3
509       51         8.3
510       91         8.4
511      153         8.5
512       83         9.0

[513 rows x 2 columns]

In [33]:
non.corr()

Runtime  IMDB Score
Runtime     1.000000   -0.055775
IMDB Score -0.055775    1.000000

In [41]:
high_score = netflix_originals.loc[netflix_originals['IMDB Score'].idxmax()]
high_score

Title         David Attenborough: A Life on Our Planet
Genre                                      Documentary
Premiere                           2020-10-04 00:00:00
Runtime                                             83
IMDB Score                                         9.0
Language                                       English
Name: 512, dtype: object

In [54]:
most_common_language = netflix_originals['Language'].value_counts().idxmax()
language_count = netflix_originals['Language'].value_counts().max()
print(f"la lengua más representada es {most_common_language} con {language_count} títulos")

la lengua más representada es English con 352 títulos


In [55]:
top_5_titles = netflix_originals.nlargest(5, 'IMDB Score')[['Title', 'IMDB Score']]
print("Los cinco títulos más puntuados:")
print(top_5_titles)

Los cinco títulos más puntuados:
                                           Title  IMDB Score
512     David Attenborough: A Life on Our Planet         9.0
511                      Springsteen on Broadway         8.5
510  Winter on Fire: Ukraine's Fight for Freedom         8.4
508                       Cuba and the Cameraman         8.3
509                       Dancing with the Birds         8.3


In [56]:
top_5_languages = netflix_originals['Language'].value_counts().head(5)
print("\nLas cinco lenguas más representadas:")
print(top_5_languages)


Las cinco lenguas más representadas:
Language
English    352
Hindi       28
Spanish     26
French      18
Italian     14
Name: count, dtype: int64


## Conclusión de Netflix_Originals
- La única consistencia en los datos fue encontrada en la columna premiere.
- No se encontraron nulos ni duplicados. 
- Es un dataframe que consiste en 513 rows × 6 columns
- Tenemos datos de tipo object y de tipo numerico (int y float)

# Análisis netflix_titles

### conclusiones
- hay valores nulos significativos en varias columnas pero se van a ignorar.
- es un dataframe de 8807 filas x 12 columnas
- hay duplicados

In [11]:
netflix_titles = netflix_titles.loc[:, ~netflix_titles.columns.str.contains('^Unnamed')]
netflix_titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [15]:
print(netflix_originals['Title'].sample(10))

193          Come Sunday
488    The Bleeding Edge
319              Bulbbul
315            Tigertail
417     Grass Is Greener
271              Birders
58       Coffee & Kareem
284      Take Your Pills
246                Benji
435         Ladies First
Name: Title, dtype: object


In [16]:
print(netflix_titles['title'].sample(10))

928            Best of the Best
2568    Hangar 1: The UFO Files
5047    The Cloverfield Paradox
4378            The Pixar Story
4509                     Soorma
5148                     Voyeur
6935              Hart of Dixie
804              Population 436
1199                    Savages
1911               Human Nature
Name: title, dtype: object


In [19]:
# Identificar títulos de netflix_originals que no se encontraron en netflix_titles
no_match = netflix_originals[~netflix_originals['Title'].isin(netflix_titles['title'])]
print(f"Títulos no encontrados en netflix_titles: {no_match.shape[0]}")
print(no_match.head())

Títulos no encontrados en netflix_titles: 0
Empty DataFrame
Columns: [Title, Genre, Premiere, Runtime, IMDB Score, Language]
Index: []


In [64]:
netflix_titles.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration        4813
listed_in          0
description        0
dtype: int64

- las columnas que más nulos tienen son director y duración

In [63]:
netflix_titles.nunique()

show_id         8807
type               2
title           8807
director        4528
cast            7692
country          748
date_added      1767
release_year      74
rating            17
duration         185
listed_in        514
description     8775
dtype: int64

- hay descripciones con duplicados


In [65]:
netflix_titles.head(10)

show_id     type                             title  \
0      s1    Movie              Dick Johnson Is Dead   
1      s2  TV Show                     Blood & Water   
2      s3  TV Show                         Ganglands   
3      s4  TV Show             Jailbirds New Orleans   
4      s5  TV Show                      Kota Factory   
5      s6  TV Show                     Midnight Mass   
6      s7    Movie  My Little Pony: A New Generation   
7      s8    Movie                           Sankofa   
8      s9  TV Show     The Great British Baking Show   
9     s10    Movie                      The Starling   

                        director  \
0                Kirsten Johnson   
1                            NaN   
2                Julien Leclercq   
3                            NaN   
4                            NaN   
5                  Mike Flanagan   
6  Robert Cullen, José Luis Ucha   
7                   Haile Gerima   
8                Andy Devonshire   
9                 Theodore Melfi   

                                                cast  \
0                                                NaN   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                                NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                             country          date_added  \
0                                      United States  September 25, 2021   
1                                       South Africa  September 24, 2021   
2                                                NaN  September 24, 2021   
3                                                NaN  September 24, 2021   
4                                              India  September 24, 2021   
5                                                NaN  September 24, 2021   
6                                                NaN  September 24, 2021   
7  United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                     United Kingdom  September 24, 2021   
9                                      United States  September 24, 2021   

   release_year rating   duration  \
0          2020  PG-13     90 min   
1          2021  TV-MA  2 Seasons   
2          2021  TV-MA        NaN   
3          2021  TV-MA        NaN   
4          2021  TV-MA  2 Seasons   
5          2021  TV-MA        NaN   
6          2021     PG        NaN   
7          1993  TV-MA    125 min   
8          2021  TV-14  9 Seasons   
9          2021  PG-13    104 min   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   
5                 TV Dramas, TV Horror, TV Mysteries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
8                       British TV Shows, Reality TV   
9                                   Comedies, Dramas   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...  
5  The arrival of a charismatic young priest brin...  
6  Equestria's divided. But a bright-eyed hero be...  
7  On a photo shoot in Ghana, an American model s...  
8  A talented 

In [61]:
netflix_titles.describe()

release_year
count   8807.000000
mean    2014.180198
std        8.819312
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

# Unión de base de datos

- Se hace un LEFT JOIN para ver qué titulos en el catálogo general (netflix_titles) no son originales de Nertflix. 

In [5]:
netflix_originals.rename(columns={'Title': 'title'}, inplace=True)
merged_data = pd.merge(netflix_titles, netflix_originals, how='left', on='title')


In [9]:
merged_data.sample(4)

Unnamed: 0 show_id     type                                      title  \
1577        1577   s1578    Movie  emicida: amarelo - it’s all for yesterday   
2680        2680   s2681    Movie                     puerto ricans in paris   
5953        5953   s5954    Movie                     joseph: king of dreams   
2659        2659   s2660  TV Show                     middleditch & schwartz   

                           director  \
1577                            NaN   
2680                    Ian Edelman   
5953  Rob LaDuca, Robert C. Ramirez   
2659                    Ryan Polito   

                                                   cast  \
1577                                            Emicida   
2680  Luis Guzmán, Edgar Garcia, Alice Taglioni, Ros...   
5953  Ben Affleck, Mark Hamill, Richard Herd, Mauree...   
2659                   Thomas Middleditch, Ben Schwartz   

                                    country          date_added  release_year  \
1577                                 Brazil    December 8, 2020          2020   
2680                          United States      April 16, 2020          2015   
5953  United States, France, United Kingdom  September 27, 2011          2000   
2659                          United States      April 21, 2020          2020   

     rating duration                                          listed_in  \
1577  TV-MA   90 min  Documentaries, International Movies, Music & M...   
2680      R   82 min                                           Comedies   
5953  TV-PG   75 min  Children & Family Movies, Dramas, Faith & Spir...   
2659  TV-MA      NaN                                        TV Comedies   

                                            description Genre Premiere  \
1577  Between scenes from his concert in São Paulo's...   NaN      NaN   
2680  New York police detectives Luis and Eddie expe...   NaN      NaN   
5953  With his gift of dream interpretation and his ...   NaN      NaN   
2659  Comedy duo Thomas Middleditch and Ben Schwartz...   NaN      NaN   

      Runtime  IMDB Score Language  
1577      NaN         NaN      NaN  
2680      NaN         NaN      NaN  
5953      NaN         NaN      NaN  
2659      NaN         NaN      NaN

In [16]:
merged_data.isnull().sum()

Unnamed: 0         0
show_id            0
type               0
Title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration        4813
listed_in          0
description        0
Genre           8294
Premiere        8294
Runtime         8294
IMDB Score      8294
Language        8294
dtype: int64

# Filtrado de datos.

In [8]:
#1. 

merged_data.loc[merged_data['type'] == 'Movie']


Unnamed: 0 show_id   type                             title  \
0              0      s1  Movie              dick johnson is dead   
6              6      s7  Movie  my little pony: a new generation   
7              7      s8  Movie                           sankofa   
9              9     s10  Movie                      the starling   
12            12     s13  Movie                      je suis karl   
...          ...     ...    ...                               ...   
8801        8801   s8802  Movie                           zinzana   
8802        8802   s8803  Movie                            zodiac   
8804        8804   s8805  Movie                        zombieland   
8805        8805   s8806  Movie                              zoom   
8806        8806   s8807  Movie                            zubaan   

                           director  \
0                   Kirsten Johnson   
6     Robert Cullen, José Luis Ucha   
7                      Haile Gerima   
9                    Theodore Melfi   
12              Christian Schwochow   
...                             ...   
8801                Majid Al Ansari   
8802                  David Fincher   
8804                Ruben Fleischer   
8805                   Peter Hewitt   
8806                    Mozez Singh   

                                                   cast  \
0                                                   NaN   
6     Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7     Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
...                                                 ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                country          date_added  \
0                                         United States  September 25, 2021   
6                                                   NaN  September 24, 2021   
7     United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
9                                         United States  September 24, 2021   
12                              Germany, Czech Republic  September 23, 2021   
...                                                 ...                 ...   
8801                       United Arab Emirates, Jordan       March 9, 2016   
8802                                      United States   November 20, 2019   
8804                                      United States    November 1, 2019   
8805                                      United States    January 11, 2020   
8806                                              India       March 2, 2019   

      release_year rating duration  \
0             2020  PG-13   90 min   
6             2021     PG      NaN   
7             1993  TV-MA  125 min   
9             2021  PG-13  104 min   
12            2021  TV-MA  127 min   
...            ...    ...      ...   
8801          2015  TV-MA      NaN   
8802          2007      R  158 min   
8804          2009      R      NaN   
8805          2006     PG      NaN   
8806          2015  TV-14  111 min   

                                             listed_in  \
0                                        Documentaries   
6                             Children & Family Movies   
7     Dramas, Independent Movies, International Movies   
9                                     Comedies, Dramas   
12                        Dramas, International Movies   
...                                                ...   
8801           Dramas, International Movies, Thrillers   
8802                    Cult Movies, Dramas, Thrillers   
8804                           Comedies, Horror Movies   
8805         

In [9]:
#2 solo titutlos y duracion de todos los contenidos que sean tipo tv show 
tv_shows = merged_data.loc[merged_data['type'] == "TV Show", ['title', 'duration']]
tv_shows

title   duration
1             blood & water  2 Seasons
2                 ganglands        NaN
3     jailbirds new orleans        NaN
4              kota factory  2 Seasons
5             midnight mass        NaN
...                     ...        ...
8795        yu-gi-oh! arc-v  2 Seasons
8796             yunus emre  2 Seasons
8797              zak storm        NaN
8800     zindagi gulzar hai        NaN
8803            zombie dumb  2 Seasons

[2676 rows x 2 columns]

In [10]:
us_titles = merged_data.loc[merged_data['country'] == "United States"]
us_titles

Unnamed: 0 show_id     type                 title         director  \
0              0      s1    Movie  dick johnson is dead  Kirsten Johnson   
9              9     s10    Movie          the starling   Theodore Melfi   
15            15     s16  TV Show     dear white people              NaN   
27            27     s28    Movie             grown ups     Dennis Dugan   
28            28     s29    Movie            dark skies    Scott Stewart   
...          ...     ...      ...                   ...              ...   
8791        8791   s8792    Movie           young adult    Jason Reitman   
8793        8793   s8794    Movie  yours, mine and ours     Raja Gosnell   
8802        8802   s8803    Movie                zodiac    David Fincher   
8804        8804   s8805    Movie            zombieland  Ruben Fleischer   
8805        8805   s8806    Movie                  zoom     Peter Hewitt   

                                                   cast        country  \
0                                                   NaN  United States   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...  United States   
15    Logan Browning, Brandon P. Bell, DeRon Horton,...  United States   
27    Adam Sandler, Kevin James, Chris Rock, David S...  United States   
28    Keri Russell, Josh Hamilton, J.K. Simmons, Dak...  United States   
...                                                 ...            ...   
8791  Charlize Theron, Patton Oswalt, Patrick Wilson...  United States   
8793  Dennis Quaid, Rene Russo, Sean Faris, Katija P...  United States   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   

              date_added  release_year rating duration  \
0     September 25, 2021          2020  PG-13   90 min   
9     September 24, 2021          2021  PG-13  104 min   
15    September 22, 2021          2021  TV-MA      NaN   
27    September 20, 2021          2010  PG-13      NaN   
28    September 19, 2021          2013  PG-13      NaN   
...                  ...           ...    ...      ...   
8791   November 20, 2019          2011      R      NaN   
8793   November 20, 2019          2005     PG      NaN   
8802   November 20, 2019          2007      R  158 min   
8804    November 1, 2019          2009      R      NaN   
8805    January 11, 2020          2006     PG      NaN   

                                 listed_in  \
0                            Documentaries   
9                         Comedies, Dramas   
15                  TV Comedies, TV Dramas   
27                                Comedies   
28         Horror Movies, Sci-Fi & Fantasy   
...                                    ...   
8791  Comedies, Dramas, Independent Movies   
8793    Children & Family Movies, Comedies   
8802        Cult Movies, Dramas, Thrillers   
8804               Comedies, Horror Movies   
8805    Children & Family Movies, Comedies   

                                            description        Genre  \
0     As her father nears the end of his life, filmm...  Documentary   
9     A woman adjusting to life after a loss contend...          NaN   
15    Students of color navigate the daily slights a...          NaN   
27    Mourning the loss of their beloved junior high...          NaN   
28    A family’s idyllic suburban life shatters when...          NaN   
...                                                 ...          ...   
8791  When a divorced writer gets a letter from an o...          NaN   
8793  When a father of eight and a mother of 10 prep...          NaN   
8802  A political cartoonist, a crime reporter and a...          NaN   
8804  Looking to survive in a world taken over by zo...          NaN   
8805  Dragged from civilian life, a former superhero...          NaN   

             Premiere  Runtime  IMDB Score Language  
0     October 2, 2020     90.0     

In [ ]:
merged_data = merged_data.loc[:, ~merged_data.columns.str.contains('^Unnamed')]


In [12]:
prod2018 = merged_data.loc[merged_data['release_year'] == 2018, ['title', 'listed_in']]
prod2018

title  \
21                               resurrection: ertugrul   
37                                          angry birds   
73                                         king of boys   
94                                            show dogs   
96    if i leave here tomorrow: a film about lynyrd ...   
...                                                 ...   
8732                                      white chamber   
8743                                           wildlife   
8774                                          يوم الدين   
8775                                    yeh meri family   
8803                                        zombie dumb   

                                              listed_in  
21    International TV Shows, TV Action & Adventure,...  
37                                Kids' TV, TV Comedies  
73                         Dramas, International Movies  
94                   Children & Family Movies, Comedies  
96                      Documentaries, Music & Musicals  
...                                                 ...  
8732                        Sci-Fi & Fantasy, Thrillers  
8743                         Dramas, Independent Movies  
8774   Dramas, Independent Movies, International Movies  
8775                International TV Shows, TV Comedies  
8803             Kids' TV, Korean TV Shows, TV Comedies  

[1147 rows x 2 columns]

In [14]:
sin_director = merged_data.loc[merged_data['director'].isna(), ['title']]
sin_director

title
1                           blood & water
3                   jailbirds new orleans
4                            kota factory
10    vendetta: truth, lies and the mafia
14        crime stories: india detectives
...                                   ...
8795                      yu-gi-oh! arc-v
8796                           yunus emre
8797                            zak storm
8800                   zindagi gulzar hai
8803                          zombie dumb

[2634 rows x 1 columns]

# Filtrado de datos con iloc.

In [16]:
sub1 = merged_data.iloc[:5, [merged_data.columns.get_loc('title'), merged_data.columns.get_loc('director'), merged_data.columns.get_loc('country'),]]
sub1

title         director        country
0   dick johnson is dead  Kirsten Johnson  United States
1          blood & water              NaN   South Africa
2              ganglands  Julien Leclercq            NaN
3  jailbirds new orleans              NaN            NaN
4           kota factory              NaN          India

In [19]:
sub2 = merged_data.iloc[-5:, [merged_data.columns.get_loc('title'), merged_data.columns.get_loc('description')]]
sub2

title                                        description
8802       zodiac  A political cartoonist, a crime reporter and a...
8803  zombie dumb  While living alone in a spooky town, a young g...
8804   zombieland  Looking to survive in a world taken over by zo...
8805         zoom  Dragged from civilian life, a former superhero...
8806       zubaan  A scrappy but poor boy worms his way into a ty...

In [21]:
sub3 = merged_data.iloc[:10, 1:5]
sub3

type                             title                       director  \
0    Movie              dick johnson is dead                Kirsten Johnson   
1  TV Show                     blood & water                            NaN   
2  TV Show                         ganglands                Julien Leclercq   
3  TV Show             jailbirds new orleans                            NaN   
4  TV Show                      kota factory                            NaN   
5  TV Show                     midnight mass                  Mike Flanagan   
6    Movie  my little pony: a new generation  Robert Cullen, José Luis Ucha   
7    Movie                           sankofa                   Haile Gerima   
8  TV Show     the great british baking show                Andy Devonshire   
9    Movie                      the starling                 Theodore Melfi   

                                                cast  
0                                                NaN  
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...  
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...  
3                                                NaN  
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...  
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...  
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...  
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...  
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...  
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...

In [23]:
sub4 = merged_data.iloc[-7:, [merged_data.columns.get_loc('title'), merged_data.columns.get_loc('director'), merged_data.columns.get_loc('country')]]
sub4

title         director                       country
8800  zindagi gulzar hai              NaN                      Pakistan
8801             zinzana  Majid Al Ansari  United Arab Emirates, Jordan
8802              zodiac    David Fincher                 United States
8803         zombie dumb              NaN                           NaN
8804          zombieland  Ruben Fleischer                 United States
8805                zoom     Peter Hewitt                 United States
8806              zubaan      Mozez Singh                         India

In [25]:
sub5 = merged_data.iloc[15:26, :4]
sub5

show_id     type                                              title  \
15     s16  TV Show                                  dear white people   
16     s17    Movie  europe's most dangerous man: otto skorzeny in ...   
17     s18  TV Show                                    falsa identidad   
18     s19    Movie                                          intrusion   
19     s20  TV Show                                             jaguar   
20     s21  TV Show    monsters inside: the 24 faces of billy milligan   
21     s22  TV Show                             resurrection: ertugrul   
22     s23    Movie                                    avvai shanmughi   
23     s24    Movie       go! go! cory carson: chrissy takes the wheel   
24     s25    Movie                                              jeans   
25     s26  TV Show                               love on the spectrum   

                                         director  
15                                            NaN  
16  Pedro de Echave García, Pablo Azorín Williams  
17                                            NaN  
18                                     Adam Salky  
19                                            NaN  
20                                Olivier Megaton  
21                                            NaN  
22                                 K.S. Ravikumar  
23                        Alex Woo, Stanley Moore  
24                                     S. Shankar  
25                                            NaN

In [33]:
sub6 = merged_data.iloc[20:31, [merged_data.columns.get_loc('title'), merged_data.columns.get_loc('release_year'), merged_data.columns.get_loc('rating')]]
sub6

title  release_year rating
20  monsters inside: the 24 faces of billy milligan          2021  TV-14
21                           resurrection: ertugrul          2018  TV-14
22                                  avvai shanmughi          1996  TV-PG
23     go! go! cory carson: chrissy takes the wheel          2021   TV-Y
24                                            jeans          1998  TV-14
25                             love on the spectrum          2021  TV-14
26                                   minsara kanavu          1997  TV-PG
27                                        grown ups          2010  PG-13
28                                       dark skies          2013  PG-13
29                                         paranoia          2013  PG-13
30                                  ankahi kahaniya          2021  TV-14